### ISSUE TITLE
Still having troubles with python@2, Mojave and no zlib

### ISSUE THREAD

**Issue by sashkab** [Homebrew/homebrew-core#34549]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

After #34473 I'm still having issues with building python@2 from source on Mojave and ending up with [non-working zlib](https://gist.githubusercontent.com/sashkab/aebbbfc7f98eba4802346d51ba531ca7/raw/821bbd582778caa1845c3930715209fbd0411497/03.make) see `Python build finished, but the necessary bits to build these modules were not found:` section.  (Full gist — https://gist.github.com/sashkab/aebbbfc7f98eba4802346d51ba531ca7)

**Comment by MikeMcQuaid** [comment]
> CLT: 10.1.0.0.1.1539992718
> Xcode: N/A

This is the most obvious difference between my machine/CI.

**Comment by sashkab** [comment]
Few notes:

1. Please note, that I have no `/usr/include` on my system as this location has been deprecated by Apple and all includes live under SDK, which can be found by running `xcrun --show-sdk-path`. 
2. I did not install `/Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg`, unless `brew doctor`
3. Debugging logic in the `setup.py` of Python 2.7.15, it seems Python does not get `/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include` path, therefore unable to locate `zlib.h` there.

**Comment by sashkab** [comment]
@MikeMcQuaid do you have Xcode installed as well in CI?

**Comment by MikeMcQuaid** [comment]
> Xcode installed as well in CI?

Yes.



> * I did not install `/Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg`.

That's not installed on my machine or in CI either.

**Comment by sashkab** [comment]
Just to avoid that my system with it's 6+years of history might be causing issues,  I spun up new VM with 10.14.1 and installed Homebrew (it installed CLT): python@2 still failure with zlib.

Logs: https://gist.github.com/8cebb7475ee6127a80f62c0810933338

Right after that, I installed Xcode, configured to use Xcode command line tools. Build resulted in fully operational python@2 with zlib. 

Logs: https://gist.github.com/a2ca51ac3529e9d826c9a56ade477acf

I kind of puzzled.

**Comment by MikeMcQuaid** [comment]
There will need to be a CLT-specific fix. You could reticle the issue accordingly. No issue with normal python (not @2)?

**Comment by sashkab** [comment]
> No issue with normal python (not @2)?

I did not notice any issues: zlib builds without a problem.

**Comment by sashkab** [comment]
So, if we pass sysroot, we need to use `-I /usr/include`, because there is logic in Python's setup.py to join sysroot to the additional include directory and then look for include file:

```python
 96  ->	        if host_platform == 'darwin' and is_macosx_sdk_path(dir):
 97  	            f = os.path.join(sysroot, dir[1:], filename)
```

**Comment by MikeMcQuaid** [comment]
@sashkab Weird but cool. Can you submit a PR?

**Comment by sashkab** [comment]
@MikeMcQuaid it got even weirder — see PR #34562

**Comment by Nicd** [comment]
This is also happening to me with youtube-dl on macOS 10.14.1:

```
Traceback (most recent call last):
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 165, in _run_module_as_main
    mod_name, loader, code, fname = _get_main_module_details(_Error)
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 133, in _get_main_module_details
    return _get_module_details(main_name)
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 121, in _get_module_details
    raise error(format(e))
ImportError: can't decompress data; zlib not available
```

**Comment by sashkab** [comment]
it's not stale.

**Comment by MikeMcQuaid** [comment]
We no longer support building from source.

**Comment by Nicd** [comment]
I have not built from source and my youtube-dl does not work as described above. Is it a different issue then?

**Comment by MikeMcQuaid** [comment]
@Nicd Yes, please open a new issue, thanks.

**Comment by fishy** [comment]
@MikeMcQuaid I think the current `python@2` formula is still building from source and having this problem in mojave. `sudo installer -pkg /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg -target / && brew reinstall python@2` fixed the issue.

**Comment by Nicd** [comment]
Just a note (I did not open that new issue), my youtube-dl has been fixed now after I updated python@2 to the latest version (there was some new update).
